### PR TITLE
MAINTAINERS file cleanup

### PR DESCRIPTION
**Pull request by icecrime**
First phase in cleaning up the MAINTAINERS file to make it better reflect reality by:
- Removing "Operators" that have no practical role.
- Removing "Subsystems" as they are separate repositories with  their own MAINTAINERS files.

Overall, the goal of this first PR is to make the file more Engine specific. All the names are preserved until we can move the single source of truth to a different repository.

Contributes to #18319.

Signed-off-by: Arnaud Porterie arnaud.porterie@docker.com

### CODE REVIEW

File: MAINTAINERS [hunk 1]
```diff
@@ -271,127 +234,18 @@ made through a pull request.
...
-				"duglin",
-				"erikh",
-				"tibor"
-			]
+    [Org."Docs maintainers"]
+
+    # TODO Describe the docs maintainers role.
+
+        people = [
+            "james",
```
Comment (thaJeztah) [inline MAINTAINERS]: oh, while you're at it, can you remove "james", because it's a duplicate of "jamtur01" :smile:

File: MAINTAINERS [hunk 2]
```diff
@@ -271,127 +234,18 @@ made through a pull request.
...
-		[Org.Subsystem.compose]
-
-			people = [
-				"aanand"
-			]
-
-		[Org.Subsystem.builder]
-
-			people = [
-				"duglin",
```
Comment (thaJeztah) [inline MAINTAINERS]: Careful; doug was never added to Core Maintainers, so better add him to that list

Comment (thaJeztah) [inline MAINTAINERS]: oh, and I see Brian also never was added to one of the lists :smile:

**Comment by jfrazelle** [comment]
LGTM

File: MAINTAINERS (hunk 2 above)
Comment (jfrazelle) [inline MAINTAINERS]: +1

File: MAINTAINERS [hunk 3]
```diff
@@ -271,127 +234,18 @@ made through a pull request.
...
-				"samalba",
-				"sday",
-				"vbatts"
-			]
-
-		[Org.Subsystems."build tools"]
-
-			people = [
-				"shykes",
-				"tianon"
```
Comment (tianon) [inline MAINTAINERS]: Does this mean I get to be moved to the `core` list instead? :innocent:

Comment (jfrazelle) [inline MAINTAINERS]: tianon is not in core list either so add him there

Comment (jfrazelle) [inline MAINTAINERS]: jynx whoa

Comment (thaJeztah) [inline MAINTAINERS]: `[Org."<3"]`

File: MAINTAINERS (hunk 2 above)
Comment (icecrime) [inline MAINTAINERS]: I noted that, but I think it's a different PR ;-)

File: MAINTAINERS (hunk 1 above)
Comment (icecrime) [inline MAINTAINERS]: Ha!

File: MAINTAINERS (hunk 2 above)
Comment (thaJeztah) [inline MAINTAINERS]: Alright, sgtm!

Comment (duglin) [inline MAINTAINERS]: erikh too

Comment (thaJeztah) [inline MAINTAINERS]: https://github.com/docker/docker/pull/18321/files#diff-bb5907b965b501e1856ccd79d2c4f642R223

Comment (duglin) [inline MAINTAINERS]: ah cool

**Comment by tiborvass** [comment]
LGTM after fixing duplicate james

**Comment by icecrime** [comment]
Ok this made no sense in many ways :-) Added @cpuguy83 @duglin @tianon in core maintainers (because: come on), and fix the dupe.

**Comment by thaJeztah** [comment]
LGTM, welcome @cpuguy83 @duglin @tianon ! :innocent:

**Comment by jfrazelle** [comment]
LGTM

**Comment by tiborvass** [comment]
I didn't even realize they weren't part of core maintainers

**Comment by jfrazelle** [comment]
Same here, the were in our hearts

**Comment by cpuguy83** [comment]
<3 LGTM

**Comment by tianon** [comment]
:metal: :heart: LGTM

**Comment by estesp** [comment]
I don't **think** this file edit is going to break windows and that CI run looks stuck, so L G T M and I'll merge.. :)

**Comment by thaJeztah** [comment]
tsk tsk

**Comment by vdemeester** [comment]
:metal: yay :heart_eyes_cat:

**Pull request merged**